src/test/java/org/jitsi/meet/test/LockRoomDigitsOnlyTest.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright @ 2015 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.meet.test;
+
+import org.jitsi.meet.test.pageobjects.web.*;
+import org.jitsi.meet.test.util.*;
+
+import org.jitsi.meet.test.web.*;
+import org.openqa.selenium.*;
+import org.testng.annotations.*;
+
+import static org.testng.Assert.*;
+
+/**
+ * Tests that the digits only password feature works.
+ *
+ * 1. Lock the room with a string (shouldn't work)
+ * 2. Lock the room with a valid numeric password (should work)
+ */
+public class LockRoomDigitsOnlyTest extends WebTestBase
+{
+    /**
+     * Stops the participant. And locks the room from participant1.
+     */
+    @Test
+    public void lockRoomWithDigitsOnly()
+    {
+        ensureOneParticipant(
+            getJitsiMeetUrl().appendConfig(
+                "config.roomPasswordNumberOfDigits=5"));
+
+        // just in case wait
+        TestUtils.waitMillis(1000);
+    
+        WebParticipant participant1 = getParticipant1();
+    
+        SecurityDialog securityDialog = participant1.getSecurityDialog();
+        securityDialog.open();
+    
+        assertFalse(securityDialog.isLocked());
+
+        // Set a non-numeric password.
+        boolean gotValidationError = false;
+        try {
+            securityDialog.addPassword("AAAAA");
+        } catch (ValidationError e) {
+            gotValidationError = true;
+        }
+        assertTrue(gotValidationError);
+    
+        TestUtils.waitMillis(1000);
+        assertFalse(securityDialog.isOpen());
+    
+        securityDialog.open();
+
+        assertFalse(securityDialog.isLocked());
+
+        // Set a valid numeric password.
+        securityDialog.addPassword("12345");
+        TestUtils.waitMillis(1000);
+        securityDialog.close();
+    
+        TestUtils.waitMillis(1000);
+        securityDialog.open();
+
+        assertTrue(securityDialog.isLocked());
```

Comment:
I would love to get rid of all the waits ... We are trying to clear those, as during the years we have a lot of those (I personally am responsible for a lot of those) some without explanation ... and it is hard to understand after some time ...

Reply:
Sounds good, will do!